Solution1/Solution1.Tests/ConsoleApp/Services/UserCommunicateServiceTests.cs

```diff
@@ -17,81 +24,177 @@ public class UserCommunicateServiceTests
+
+            _invokerMock.Verify(i => i.RunAsync(It.IsAny<CurrentWeatherCommand>()));
             Assert.Equal(expected, consoleOutput.ToString());
         }
 
+        [Fact]
+        public async Task CloseApplication_Success()
+        {
+            // Arrange
+            var pattern = $"Сlose the application{Environment.NewLine}$";
+            
+            var consoleOutput = new StringWriter();
+            Console.SetOut(consoleOutput);
+            Console.SetIn(new StringReader(string.Format("0{0}", Environment.NewLine)));
+
+            //Act
+            await _userCommunicationService.CommunicateAsync();
+
+            //Assert
+            Assert.Matches(pattern, consoleOutput.ToString());;
+        }
+
         [Theory]
-        [InlineData("Minsk", "Please, enter city name:\r\nUnexpected error. Try one time yet.\r\n", null)]
-        [InlineData("Minsk", "Please, enter city name:\r\nRequest error. Try again later.\r\n", HttpStatusCode.BadRequest)]
-        [InlineData("AAA", "Please, enter city name:\r\nEntered incorrect city name. Try one time yet.\r\n", HttpStatusCode.NotFound)]
-        public async Task Communicate_EnterCityName_HandlingExceptionAndShowNoticeAsync(string cityName, string message, HttpStatusCode? statusCode)
+        [InlineData("Minsk", "Unexpected error. Try one time yet.\r\n", null, false)]
+        [InlineData(default, "Validation error.", null, true)]
+        [InlineData("Minsk", "Request error. Try again later.\r\n", HttpStatusCode.BadRequest, false)]
+        [InlineData("AAA", "Entered incorrect city name. Try one time yet.\r\n", HttpStatusCode.NotFound, false)]
+        public async Task Communicate_EnterCityName_HandlingExceptionAndShowNoticeAsync(string cityName, string message, HttpStatusCode? statusCode, bool IsValidateEror)
         {
             // Arrange
-            var exception = statusCode.HasValue ? new HttpRequestException(null, null, statusCode) : new Exception();
+            var exception = statusCode.HasValue ? new HttpRequestException(null, null, statusCode) :
+                            IsValidateEror ? new ValidationException(new List<ValidationFailure>() { new ValidationFailure("CityName", message) }) : new Exception();
 
-            _weatherServiceMock
-                .Setup(weatherApiService =>
-                    weatherApiService
-                    .GetByCityNameAsync(cityName))
+            _invokerMock
+                .Setup(invoker =>
+                    invoker
+                    .RunAsync(It.IsAny<CurrentWeatherCommand>()))
                 .Throws(exception);
 
             var consoleOutput = new StringWriter();
             Console.SetOut(consoleOutput);
+            Console.SetIn(new StringReader(string.Format("1{0}{1}{0}", Environment.NewLine, cityName)));
+
+            //Act
+            await _userCommunicationService.CommunicateAsync();
+
+            //Assert
+            var expected = $"{Menu.GetMenuRepresentation()}" + Environment.NewLine +
```

Comment:
interpolation and pluses at the same time....

Reply:
below in other files same situation

Reply:
+